server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -275,6 +285,47 @@ public abstract class BlobStoreRepository extends AbstractLifecycleComponent imp
         Setting.Property.Deprecated
     );
 
+    /**
+     * Sets the cache size for snapshot repository data, defaulting to 1% of node heap memory. The maximum is 5%.
+     */
+    public static final Setting<ByteSizeValue> SNAPSHOT_REPOSITORY_DATA_CACHE_THRESHOLD = new Setting<>(
+        SNAPSHOT_REPOSITORY_DATA_CACHET_THRESHOLD_SETTING_NAME,
+        calculateDefaultSnapshotRepositoryDataCacheThreshold(),
+        (s) -> {
+            ByteSizeValue userDefinedLimit = parseBytesSizeValueOrHeapRatio(s, SNAPSHOT_REPOSITORY_DATA_CACHET_THRESHOLD_SETTING_NAME);
+            ByteSizeValue jvmHeapSize = JvmInfo.jvmInfo().getMem().getHeapMax();
+
+            if ((userDefinedLimit.getKbFrac() / jvmHeapSize.getKbFrac()) > (SNAPSHOT_REPOSITORY_DATA_CACHE_THRESHOLD_MAX_PERCENTAGE
```

Comment:
I think we don't need this SNAPSHOT_REPOSITORY_DATA_CACHE_THRESHOLD_MAX_PERCENTAGE:
 - the min is 500k (as before)
 - the max (for cache) is 1% of the heap 

So for small heaps, we would be fine, but it is essentially ~100x increase for regularly sized clusters (4Gb/8Gb).

Reply:
Hello @reta, there is a part I am not fully understanding, so I would like to ask for clarification. 😄
Are you suggesting that we do not need to provide a maximum value for the setting that allows users to decide the threshold? From the previous discussions, I understood that this setting would be provided to users. Additionally, if users can set the percentage, I believe it was discussed that we should ensure it is not set too high.

> @inpink Assuming we have consensus with @reta's proposed plan, I'd rename the setting to be something like "snapshot.repository_metadata.cache.threshold" and add a comment describing it with something like:

Please let me know if there is any part I might be misunderstanding. I would greatly appreciate it! :)

Reply:
> Are you suggesting that we do not need to provide a maximum value for the setting that allows users to decide the threshold? 

We do have a bound, it is: `max(500kb, 1% heap)`. So the `snapshot.repository_metadata.cache.threshold` is the size, it could be any value in this range.

Reply:
Ah, I understand now! Thank you for the clear explanation. 😄🎉 I have modified the code as per your guidance.

Reply:
I believed that a static setting, rather than dynamic, was more suitable here to avoid further amplifying the unpredictability of SoftReference.

If you have any further feedback, I would greatly appreciate it!